README.md

```diff
@@ -17,49 +17,205 @@ information](LICENSE), as well as [pre-commit hooks](https://pre-commit.com)
 and [GitHub Actions](https://github.com/features/actions) configurations
 appropriate for docker containers and the major languages that we use.
 
-## Usage ##
+## Running ##
 
-### Install ###
+### Using Docker ###
 
-Pull `cisagov/example` from the Docker repository:
+```console
+docker run cisagov/example:0.0.1
+```
 
-    docker pull cisagov/example
+## Using a Docker composition ###
 
-Or build `cisagov/example` from source:
+1. Create a `docker-compose.yml` file similar to the one below.
 
-    git clone https://github.com/cisagov/skeleton-docker.git
-    cd skeleton-docker
-    docker-compose build --build-arg VERSION=0.0.1
+    ```yaml
+    version: "3.8"
 
-### Run ###
+    services:
+      example:
+        image: cisagov/example:0.0.1
+        volumes:
+          - type: bind
+            source: <your_log_dir>
+            target: /var/log
+        environment:
+          - ECHO_MESSAGE="Hello from docker-compose"
+        ports:
+          - target: "8080"
+            published: "8080"
+            protocol: tcp
+    ```
 
-    docker-compose run --rm example
+1. Start the container and detach:
 
-## Ports ##
+    ```console
+    docker-compose up --detach
+    ```
 
-This container exposes the following ports:
+## Using secrets ##
 
-| Port  | Protocol | Service  |
-|-------|----------|----------|
-| 8080  | TCP      | http     |
+This container also supports passing sensitive values via [Docker
+secrets](https://docs.docker.com/engine/swarm/secrets/).  Passing sensitive
+values like your credentials can be more secure using secrets than using
+environment variables.  See the
+[secrets](#secrets) section below for a table of all supported secret files.
 
-## Environment Variables ##
+1. To use secrets, create a `quote.txt` file containing the values you want set:
 
-| Variable      | Default Value                 | Purpose      |
-|---------------|-------------------------------|--------------|
-| ECHO_MESSAGE  | `Hello World from Dockerfile` | Text to echo |
+    ```text
+    Better lock it in your pocket.
+    ```
 
-## Secrets ##
+1. Then add the secret to your `docker-compose.yml` file:
+
+    ```yaml
+    version: "3.8"
+
+    secrets:
+      quote_txt:
+        file: quote.txt
+
+    services:
+      example:
+        image: cisagov/example:0.0.1
+        volumes:
+          - type: bind
+            source: <your_log_dir>
+            target: /var/log
+        environment:
+          - ECHO_MESSAGE="Hello from docker-compose"
+        ports:
+          - target: "8080"
+            published: "8080"
+            protocol: tcp
+        secrets:
+          - source: quote_txt
+            target: quote.txt
+    ```
+
+## Updating ##
+
+### Docker-compose ###
+
+1. Pull the new image from Docker hub:
```

Comment:
In the Docker directions you specify stopping the container so should there be a `docker-compose down` in the compose section?

Reply:
docker-compose is a bit more magical.  It will determine which services need to be brought down and back up based on which images have been updated, and their dependents.  So a `pull` and `up` will "do the right thing™". 

https://docs.docker.com/compose/reference/pull/

https://docs.docker.com/compose/reference/up/

> If there are existing containers for a service, and the service’s configuration or image was changed after the container’s creation, `docker-compose up` picks up the changes by stopping and recreating the containers (preserving mounted volumes). To prevent Compose from picking up changes, use the `--no-recreate` flag.

Reply:
mmmmmagic 👍🏻  coolio